src/Database/Type/FloatType.php

```diff
@@ -65,18 +65,18 @@ public function __construct($name = null)
      *
      * @param string|resource $value The value to convert.
      * @param \Cake\Database\Driver $driver The driver instance to convert with.
-     * @return string|resource
+     * @return string|null
      */
     public function toDatabase($value, Driver $driver)
     {
         if ($value === null || $value === '') {
             return null;
         }
         if (is_array($value)) {
-            return 1;
+            return '1';
         }
 
-        return (float)$value;
+        return sprintf('%F', (float)$value);
```

Comment:
The exponential notation would work if the field is declared as `float` not `decimal`. The root cause of this issue is that mapping `decimal` to `FloatType`. If we map `decimal` to `StringType`, `$v` in your example will be a `string` instead of `float`. So the reported issue wouldn't happen.

Using `sprintf()` might fix #9424, if the reporter doesn't store number more than 16 digits into the field. But wouldn't it introduce another issue, in some edge case?
```php
$v = 9.9999999;
echo sprintf('%F', $v);
// 10.000000
```